### PR TITLE
PLANET-7055 Setup Playwright

### PR DESCRIPTION
**Pull request by mleray**
### Description

See [PLANET-7055](https://jira.greenpeace.org/browse/PLANET-7055)

**Related PR:** https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/1001

### Testing

On local, run `npm install` to install Playwright and its needed dependencies, and then to run the tests you can use the following commands:
- `npx playwright test` for all tests
- `npx playwright test tests/e2e/404.spec.js` for individual tests

### CODE REVIEW

**Comment by planet-4** [comment]
### Test instance is ready :rocket:

:new_moon: [iocaste](https://www-dev.greenpeace.org/test-iocaste) | [admin](https://www-dev.greenpeace.org/test-iocaste/wp-admin/) | [blocks report](https://www-dev.greenpeace.org/test-iocaste/wp-admin/?page=plugin_blocks_report) | [CircleCI](https://app.circleci.com/pipelines/github/greenpeace/planet4-test-iocaste) | [composer-local.json](https://github.com/greenpeace/planet4-test-iocaste/blob/main/composer-local.json)

:watch: 2023.02.03 09:49:16

File: tests/e2e/404.spec.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  const response = await page.goto('/thispagereallywillnotexist');
+
+  // Check the page status.
+  expect(response.status()).toEqual(404);
+
+  // Check the page text.
+  const settingsText = await page.evaluate('window.p4bk_vars.page_text_404');
+  await expect(settingsText).toBeDefined();
+  const pageContent = await page.locator('.speech-bubble').innerHTML();
+  await expect(pageContent).toContain(settingsText.replace('\r\n', '\n'));
```
Comment (mleray) [inline tests/e2e/404.spec.js:13]: This wasn't working without `replace('\r\n', '\n')`, but there might be a better way to do this 🤔

**Review by lithrel — APPROVED** [review]
:+1:

File: .gitignore [hunk 2]
```diff
@@ -6,3 +6,8 @@ tests/backstop/
 
 # Built css and js assets
 assets/build/*
+
+# Playwright
+/e2e-results/
+/e2e-report/
+/playwright/.cache/
```
Comment (lithrel) [inline .gitignore:13]: do we still need the playwright directory here ?

File: tests/e2e/404.spec.js (hunk 1 above)
Comment (lithrel) [reply tests/e2e/404.spec.js:13]: Tested with international data, `.replace(/[\r]/g, '')` seems to work better

File: .gitignore (hunk 2 above)
Comment (mleray) [reply .gitignore:13]: Good question, I'm not really sure what it's for, it's added there by default during setup but I can't find anything about it in the docs 🤷‍♀️

File: tests/e2e/404.spec.js (hunk 1 above)
Comment (mleray) [reply tests/e2e/404.spec.js]: Updated! 🙂

**Review by lithrel — APPROVED** [review]
:+1:

**Pull request merged**